src/ripple/app/misc/impl/Manifest.cpp

```diff
@@ -27,44 +27,132 @@
+        if (st.isFieldPresent(sfVersion) && st.getFieldU16(sfVersion) != 0)
+            return boost::none;
+
         auto const pk = st.getFieldVL (sfPublicKey);
+
         if (! publicKeyType (makeSlice(pk)))
             return boost::none;
 
-        auto const opt_seq = get (st, sfSequence);
-        auto const opt_msig = get (st, sfMasterSignature);
-        if (!opt_seq || !opt_msig)
-            return boost::none;
+        Manifest m;
+        m.serialized.assign(reinterpret_cast<char const*>(s.data()), s.size());
+        m.masterKey = PublicKey(makeSlice(pk));
+        m.sequence = st.getFieldU32 (sfSequence);
+
+        // N.B.: Changes to the rules we use to validate the "domain" field
+        // are, implicitly, transaction breaking since serialized manifests
+        // are parsed during the execution of the LoadManifest transaction.
+        if (st.isFieldPresent(sfDomain))
+        {
+            auto const d = st.getFieldVL(sfDomain);
+
+            // A valid domain for a validator must be between 4 and 128
+            // characters long and must contain at least one . character.
+            if (boost::algorithm::clamp(d.size(), 4, 128) != d.size())
+                return boost::none;
+
+            m.domain.assign (reinterpret_cast<char const*>(d.data()), d.size());
+
+            // This regular expression should do a decent job of weeding out
+            // obviously wrong domain names but it isn't perfect. It does not
+            // really support IDNs. If this turns out to be an issue, a more
+            // thorough regex can be used or this check can just be removed.
+            static boost::regex const re(
+                "^"                     // Beginning of line
+                "("                     // Beginning of a segment
+                "(?!-)"                 //  - must not begin with '-'
+                "[a-zA-Z0-9-]{1,63}"    //  - only alphanumeric and '-'
+                "(?<!-)"                //  - must not end with '-'
+                "\\."                   // segment separator
+                ")+"                    // 1 or more segments
+                "[A-Za-z]{2,63}"        // TLD
+                "$"                     // End of line
+                , boost::regex_constants::optimize);
+
+            if (!boost::regex_match(m.domain, re))
+                return boost::none;
+        }
+
+        bool const hasEphemeralKey = st.isFieldPresent(sfSigningPubKey);
+        bool const hasEphemeralSig = st.isFieldPresent(sfSignature);
 
-        // Signing key and signature are not required for
-        // master key revocations
-        if (*opt_seq != std::numeric_limits<std::uint32_t>::max ())
+        if (m.revoked())
         {
-            auto const spk = st.getFieldVL (sfSigningPubKey);
-            if (! publicKeyType (makeSlice(spk)))
+            // Revocation manifests should not specify a new signing key
+            // or a signing key signature.
```

Comment:
It was by design; the old behavior was incorrect (what purpose does it serve to have a revoked manifest have an ephemeral signing key?)

However, it may be that we have to “accept” such manifests anyways, as there may already be some, out there in the wild.